src/funnels/automation/StableSwapper.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface SwapperLike {
+    function swap(address src, address dst, uint256 amt, uint256 minOut, address callee, bytes calldata data) external returns (uint256 out);
+}
+
+contract StableSwapper {
+    mapping (address => uint256) public wards;                           // admin
+    mapping (address => uint256) public buds;                            // facilitators
+    mapping (address => uint256) public bots;                            // whitelisted keepers
+    mapping (address => mapping (address => PairConfig)) public configs; // configs[src][dst].count is the remaining number of times that a src-to-dst swap can be performed by keepers
+                                                                         // configs[src][dst].lot    [token weis] is the amount swapped by keepers from src to dst every hop
+                                                                         // configs[src][dst].reqOut [token weis] is the minimum output amount to insist on in the swap form src to dst
+
+    address public immutable swapper;                                    // Swapper for this StableSwapper
+
+    event Rely   (address indexed usr);
+    event Deny   (address indexed usr);
+    event Kissed (address indexed usr);
+    event Dissed (address indexed usr);
```

Comment:
Should probably be `Kiss` and `Diss` as other event names are not in past tense.

Reply:
Done in https://github.com/makerdao/dss-allocator/pull/15.